qemu/tests/numa_prealloc_threads.py

```diff
@@ -0,0 +1,68 @@
+import re
+
+from avocado.utils import cpu
+from virttest import env_process
+from virttest import error_context
+from virttest.qemu_monitor import QMPCmdError
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    numa_prealloc_threads test
+    1) Boot a guest with thread-context and cpu-affinity option
+    2) Obtain the thread-id
+    3) Check the affinity obtained from QEMU is correct
+    4) With sandbox enabled, try to change the cpu-affinity
+       and handle the error
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    test.log.info("Check host CPUs number")
+    host_cpus = int(cpu.online_count())
+    smp_fixed = params.get_numeric("smp_fixed")
+    if host_cpus < smp_fixed:
+        test.cancel("The host only has %d CPUs, it needs at least %d!" % (host_cpus, smp_fixed))
+
+    params['not_preprocess'] = "no"
+    timeout = params.get_numeric("login_timeout", 1000)
+    env_process.preprocess(test, params, env)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    vm.wait_for_login(timeout=timeout)
+
+    device = "thread_context-" + params["vm_thread_contexts"]
+    error_msg = params.get("sandbox_error_message", "")
+
+    thread_id = vm.monitor.qom_get(device, "thread-id")
+    if not thread_id:
+        test.fail("No thread-id setted.")
+
+    test_affinity = params.get("vm_thread_context_cpu-affinity")
```

Comment:
Same, it is `thread_context_device.get_param('cpu-affinity')`

Reply:
Updated now, thanks!